src/smoosh/test/smoosh_priority_queue_tests.erl

```diff
@@ -0,0 +1,169 @@
+-module(smoosh_priority_queue_tests).
+
+-include_lib("proper/include/proper.hrl").
+-include_lib("couch/include/couch_eunit.hrl").
+
+-define(PROP_PREFIX, "prop_").
+
+-define(CAPACITY, 3).
+
+-define(RANDOM_FILE, lists:flatten(io_lib:format("~p", [erlang:timestamp()]))).
+
+setup() ->
+    Ctx = test_util:start_couch(),
+    Ctx.
+
+teardown(Ctx) ->
+    test_util:stop_couch(Ctx).
+
+smoosh_priority_queue_test_() ->
+    {
+        "smoosh priority queue test",
+        {
+            setup,
+            fun setup/0,
+            fun teardown/1,
+            [
+                fun prop_inverse_test_/0,
+                fun no_halt_on_corrupted_file_test/0,
+                fun no_halt_on_missing_file_test/0
+            ]
+        }
+    }.
+
+%% ==========
+%% Tests
+%% ----------
+
+%% define all tests to be able to run them individually
+prop_inverse_test_() ->
+    ?_test(begin
+        test_property(prop_inverse)
+    end).
+
+no_halt_on_corrupted_file_test() ->
+    ?_test(begin
+        Name = ?RANDOM_FILE,
+        Q0 = smoosh_priority_queue:new(Name),
+        Q = smoosh_priority_queue:open(Q0),
```

Comment:
If I am reading the code correctly this call does nothing. 

Extracts from `src/smoosh/src/smoosh_priority_queue.erl`
```
new(Name) ->
    #priority_queue{name = Name, map = maps:new(), tree = gb_trees:empty()}.
 
open(#priority_queue{name = Name} = Q) ->
    case do_open(file_name(Q)) of
        {ok, Terms} ->
           ...
        error ->
            Q
    end.

do_open(FilePath) ->
    case file:read_file(FilePath) of
        {ok, Content} ->
              ...
        {error, enoent} ->
            couch_log:notice(
                "~p (~p) Queue file ~s does not exist. Not restoring.", [?MODULE, enoent, FilePath]
            ),
            error;
    end.        
```

Reply:
I am fine to keep it. Just wanted to clarify my understanding.

Reply:
Yes, that is correct. The first call to `smoosh_priority_queue:open/1` has no effect. I will remove it, no point in keeping it.